db-engine/src/test/java/com/splicemachine/db/impl/sql/SerializationTest.java

```diff
@@ -0,0 +1,156 @@
+            this.name = name;
+            this.sourceTableName = sourceTableName;
+            this.sourceSchemaName = sourceSchemaName;
+            this.iColPos = iColPos;
+            this.type = type;
+            this.bAutoIncrement = bAutoIncrement;
+            this.bUpdateableByCursor = bUpdateableByCursor;
+            this.bHasGenerationClause = bHasGenerationClause;
+        }
+
+        @Override
+        public DataTypeDescriptor getType() {
+            return type;
+        }
+
+        @Override
+        public StructField getStructField() {
+            Assert.fail("not expected call of getStructField");
+            return null;
+        }
+
+        @Override
+        public String getName() {
+            return name;
+        }
+
+        @Override
+        public String getSourceSchemaName() {
+            return sourceSchemaName;
+        }
+
+        @Override
+        public String getSourceTableName() {
+            return sourceTableName;
+        }
+
+        @Override
+        public boolean updatableByCursor() {
+            return bUpdateableByCursor;
+        }
+
+        @Override
+        public int getColumnPosition() {
+            return iColPos;
+        }
+
+        @Override
+        public boolean isAutoincrement() {
+            return bAutoIncrement;
+        }
+
+        @Override
+        public boolean hasGenerationClause() {
+            return bHasGenerationClause;
+        }
+
+        public static GenericColumnDescriptor sample1() {
+            return new GenericColumnDescriptor(new CustomResultColumnDescriptor(
+                    "name", "tabName", "schemaName", 3,
+                    new DataTypeDescriptor(TypeId.getBuiltInTypeId(Types.DECIMAL), true, 10),
+                    true, true, true)
+            );
+        }
+        public static GenericColumnDescriptor sample2() {
+            return new GenericColumnDescriptor(new CustomResultColumnDescriptor(
+                    "sample2name", "atable", "SPLICE", 9,
+                    new DataTypeDescriptor(TypeId.getBuiltInTypeId(Types.DOUBLE), false, 5),
+                    false, false, false)
+            );
+        }
+
+        public static void assertEquals(ResultColumnDescriptor rcd1, ResultColumnDescriptor rcd2)
+        {
+            Assert.assertEquals(rcd1.getSourceSchemaName(), rcd2.getSourceSchemaName());
+            Assert.assertEquals(rcd1.getSourceTableName(), rcd2.getSourceTableName());
+            Assert.assertEquals(rcd1.getName(), rcd2.getName());
+
+            Assert.assertEquals(rcd1.getType(), rcd2.getType());
+            Assert.assertEquals(rcd1.getColumnPosition(), rcd2.getColumnPosition());
+            Assert.assertEquals(rcd1.isAutoincrement(), rcd2.isAutoincrement());
+            Assert.assertEquals(rcd1.updatableByCursor(), rcd2.updatableByCursor());
+
+            // check this in DB-10582
+            // Assert.assertEquals(rcd1.hasGenerationClause(), rcd2.hasGenerationClause());
+        }
```

Comment:
Should this be uncommented?

Reply:
uncommenting it will result in a test failure. addressing this is not part of this PR, but shouldn't be ignored, that's why i created https://splicemachine.atlassian.net/browse/DB-10582 .